src/Yaapii.Atoms/Text/SplitText.cs

```diff
@@ -40,56 +40,69 @@ public sealed class SplitText : IEnumerable<String>
         private readonly IText _regex;
+        private readonly bool _remEmpty;
 
         /// <summary>
         /// A <see cref="IText"/> which has been splitted at the given string.
         /// </summary>
         /// <param name="text">text to split</param>
         /// <param name="rgx">regex to use for splitting</param>
-        public SplitText(String text, String rgx) :
-            this(
-                new TextOf(text),
-                new TextOf(rgx)
-            )
+        /// <param name="remEmpty">switch to remove empty stirngs from result or not</param>
+        public SplitText(String text, String rgx, bool remEmpty = true) : this(
+            new TextOf(text),
+            new TextOf(rgx),
+            remEmpty)
         { }
 
         /// <summary>
         /// A <see cref="IText"/> which has been splitted at the given string.
         /// </summary>
         /// <param name="text">text to split</param>
         /// <param name="rgx">regex to use for splitting</param>
-        public SplitText(String text, IText rgx) :
-            this(new TextOf(text), rgx)
+        /// <param name="remEmpty">switch to remove empty stirngs from result or not</param>
+        public SplitText(String text, IText rgx, bool remEmpty = true) : this(
+            new TextOf(text),
+            rgx,
+            remEmpty)
         { }
 
         /// <summary>
         /// A <see cref="IText"/> which has been splitted at the given string.
         /// </summary>
         /// <param name="text">text to split</param>
         /// <param name="rgx">regex to use for splitting</param>
-        public SplitText(IText text, String rgx) :
-            this(text, new TextOf(rgx))
+        /// <param name="remEmpty">switch to remove empty stirngs from result or not</param>
+        public SplitText(IText text, String rgx, bool remEmpty = true) : this(
+            text,
+            new TextOf(rgx),
+            remEmpty)
         { }
 
         /// <summary>
         /// A <see cref="IText"/> which has been splitted at the given string.
         /// </summary>
         /// <param name="text">text to split</param>
         /// <param name="rgx">regex to use for splitting</param>
-        public SplitText(IText text, IText rgx)
+        /// <param name="remEmpty">switch to remove empty stirngs from result or not</param>
+        public SplitText(IText text, IText rgx, bool remEmpty = true)
         {
             this._origin = text;
             this._regex = rgx;
+            this._remEmpty = remEmpty;
         }
 
         public IEnumerator<String> GetEnumerator()
         {
-            return new Filtered<String>(
+            var splitted =
+                new EnumerableOf<String>(
+                    new Regex(this._regex.AsString()).Split(this._origin.AsString()));
+
+            return 
+                this._remEmpty 
+                ? new Filtered<String>(
                     (str) => !String.IsNullOrEmpty(str),
```

Comment:
I think we should use ```String.IsNullOrWhiteSpace()```, don't we?

Reply:
A snippet with a white space is a relevant information for me. I would suggest to keep white spaces

Reply:
I think in that case we should rename remEmpty parameter to remBlank.
When we need the blank ones we can make a filtered and filter by String.IsNullOrEmpty.
The use case with blanks is needed more often

Reply:
Yes, I'm in.